### PR TITLE
feat: REST API: Implicit Invitation

### PR DESCRIPTION
**Pull request by sandrask**
Add implicit Invitation endpoint to did-exchange REST API.

Closes: #897

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: pkg/restapi/operation/didexchange/didexchange.go [hunk 1]
```diff
@@ -31,15 +31,16 @@ import (
...
-	receiveInvitationPath   = operationID + "/receive-invitation"
-	acceptInvitationPath    = operationID + "/{id}/accept-invitation"
-	connections             = operationID
-	connectionsByID         = operationID + "/{id}"
-	acceptExchangeRequest   = operationID + "/{id}/accept-request"
-	removeConnection        = operationID + "/{id}/remove"
-	connectionsWebhookTopic = "connections"
+	operationID                  = "/connections"
+	createInvitationPath         = operationID + "/create-invitation"
+	createImplicitInvitationPath = operationID + "/create-implicit-invitation"
```
Comment (talwinder50) [inline pkg/restapi/operation/didexchange/didexchange.go]: we only added "/create-implicit-invitation" but its showing all the const are re added. May be extra space issue

File: test/bdd/pkg/didexchange/didexchange_controller_steps.go [hunk 2]
```diff
@@ -28,18 +28,19 @@ import (
...
-	receiveInvtiationPath = connOperationID + "/receive-invitation"
-	acceptInvitationPath  = connOperationID + "/%s/accept-invitation?public=%s"
-	acceptRequestPath     = connOperationID + "/%s/accept-request?public=%s"
-	connectionsByID       = connOperationID + "/{id}"
-	createPublicDIDPath   = vdriOperationID + "/create-public-did"
-	checkForTopics        = "/checktopics"
-	publicDIDCreateHeader = `{"alg":"","kid":"","operation":"create"}`
-	sideTreeURL           = "${SIDETREE_URL}"
-	timeoutWaitForDID     = 10 * time.Second
+	connOperationID              = "/connections"
```
Comment (talwinder50) [inline test/bdd/pkg/didexchange/didexchange_controller_steps.go]: same as above

File: test/bdd/pkg/didexchange/didexchange_sdk_steps.go [hunk 3]
```diff
@@ -297,7 +297,7 @@ func resolveDID(vdriRegistry vdriapi.Registry, did string, maxRetry int) (*diddo
 }
 
 // RegisterSteps registers did exchange steps
-func (d *SDKSteps) RegisterSteps(s *godog.Suite) {
+func (d *SDKSteps) RegisterSteps(s *godog.Suite) { //nolint dupl
```
Comment (talwinder50) [inline test/bdd/pkg/didexchange/didexchange_sdk_steps.go]: just  for info what warning, linter is throwing here ??

**Comment by talwinder50** [comment]
just minor comment, it looks good

File: test/bdd/pkg/didexchange/didexchange_sdk_steps.go (hunk 3 above)
Comment (sandrask) [reply test/bdd/pkg/didexchange/didexchange_sdk_steps.go]: Linter thinks that SDK steps and Controller steps are the same. It is false alert. They are not the same. @rolsonquadras  had similar problem earlier.

File: pkg/restapi/operation/didexchange/didexchange.go (hunk 1 above)
Comment (sandrask) [reply pkg/restapi/operation/didexchange/didexchange.go]: go format

**Pull request merged**